devdoc/interlocked_requirements.md

```diff
@@ -0,0 +1,365 @@
+
+**SRS_INTERLOCKED_43_033: [** `interlocked_and` shall return the initial value of `*destination`.**]**
+
+## interlocked_and_16
+
+```c
+MOCKABLE_FUNCTION(, int16_t, interlocked_and_16, volatile int16_t*, destination, int16_t, value);
+```
+
+**SRS_INTERLOCKED_43_003 [** `interlocked_and_16` shall perform an atomic bitwise AND operation on the 16-bit integer values `*destination` and `value` and store the result in `*destination.`**]**
+
+**SRS_INTERLOCKED_43_034: [** `interlocked_and_16` shall return the initial value of `*destination`.**]**
+
+## interlocked_and_64
+
+```c
+MOCKABLE_FUNCTION(, int64_t, interlocked_and_64, volatile int64_t*, destination, int64_t, value);
+```
+
+**SRS_INTERLOCKED_43_004 [** `interlocked_and_64` shall perform an atomic bitwise AND operation on the 64-bit integer values `*destination` and `value` and store the result in `*destination`.**]**
+
+**SRS_INTERLOCKED_43_035: [** `interlocked_and_64` shall return the initial value of `*destination`.**]**
+
+
+## interlocked_and_8
+
+```c
+MOCKABLE_FUNCTION(, int8_t, interlocked_and_8, volatile int8_t*, destination, int8_t, value);
+```
+
+**SRS_INTERLOCKED_43_005 [** `interlocked_and_8` shall perform an atomic bitwise AND operation on the 8-bit integer values `*destination` and `value` and store the result in `*destination`**]**
+
+**SRS_INTERLOCKED_43_036: [** `interlocked_and_8` shall return the initial value of `*destination`. **]**
+
+## interlocked_compare_exchange
+
+```c
+MOCKABLE_FUNCTION(, int32_t, interlocked_compare_exchange, volatile int32_t*, destination, int32_t, exchange, int32_t, comperand);
+```
+
+**SRS_INTERLOCKED_43_006 [** `interlocked_compare_exchange` shall compare the 32-bit integers pointed to by `destination` and `comperand`. If they are equal, `*destination` is set to `exchange`. These operations are performed atomically.***]**
+
+**SRS_INTERLOCKED_43_037: [** `interlocked_compare_exchange` shall return the initial value of the `destination` parameter.**]**
+
+## interlocked_compare_exchange_128
+
+```c
+MOCKABLE_FUNCTION(, bool, interlocked_compare_exchange_128, volatile int64_t*, destination, int64_t, exchange_high, int64_t, exchange_low, int64_t*, comperand_result);
+```
+
+**SRS_INTERLOCKED_43_007 [** `interlocked_compare_exchange_128` shall compare `*destination` with `*comperand_result`. If they are equal, `exchange_high` and `exchange_low` are stored in the array specified by `destination`. These operations are performed atomically. The `destination` and `comperand_result` parameters are arrays of two 64-bit integers considered as a 128-bit fields.**]**
+
+**SRS_INTERLOCKED_43_039: [** `interlocked_compare_exchange_128` shall store the initial value of `*destination` in `*comperand_result` regardless of the result of the comparison.` **]**
+
+**SRS_INTERLOCKED_43_038: [** `interlocked_compare_exchange_128` shall return `1` if `*comperand_result` equals the original value of `*destination` and `0` otherwise**]**
```

Comment:
>`interlocked_compare_exchange_128` shall return `1` if `*comperand_result` equals the original value of `*destination` and `0` otherwise** [](start = 30, length = 138)

You should split this (also it returns `bool` so `true` and `false` rather than `1` and `0`):

If `comperand_result` equals the original value of the `destination` parameter then `interlocked_compare_exchange_128` shall return `true`.

If `comperand_result` does not equal the original value of the `destination` parameter then `interlocked_compare_exchange_128` shall return `false`.